### PR TITLE
docs: add mappers docs guide

### PR DESCRIPTION
**Pull request by pnadolny13**
Related to https://github.com/meltano/meltano/issues/6412#issuecomment-1471684404

Adds a guide to the docs that includes both standalone mappers and SDK based inline stream maps

- I sourced a lot of this information from https://sdk.meltano.com/en/latest/stream_maps.html and some examples from https://github.com/MeltanoLabs/meltano-map-transform/pull/117
- I put a couple examples to give a preview of what they do but I mostly wanted to link out to other docs like https://github.com/MeltanoLabs/meltano-map-transform/pull/117 and https://sdk.meltano.com/en/latest/stream_maps.html
- I felt like it was worth highlighting the difference between the two mapping methods and when you would use each, with tradeoffs. So let me know if I got anything wrong.
- Its a bit tricky to figure out the best way to refer to the two different approach. I went with "standalone mapper plugins" and "SDK 

Let me know if you have any obvious use cases I missed and what you think in general.

### CODE REVIEW

File: docs/src/_guide/mappers.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+# Features
+
+The mapper features can be viewed in 2 categories: stream level or property (record) level.
+Stream level features allow you to modify the stream itself, whereas the property level features allow you to modify contents of the stream.
+
+## Stream Level
+
+- Aliasing: streams can be aliased to provide custom naming downstream.
+- Filtering: streams records can be filtered based on any user-defined logic.
```
Comment (tayloramurphy) [inline docs/src/_guide/mappers.md:21]:
```suggestion
- Filtering: stream records can be filtered based on any user-defined logic.
```

With the way we've got property level changes labeled as `property (record)` this sentence is suddenly very confusing to me. Would `property (column)` be more appropriate?

File: docs/src/_guide/mappers.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        issues:
+          repo: record['repo'].lower()
+```
+
+For more details and examples refer to the [SDK documentation](https://sdk.meltano.com/en/latest/stream_maps.html).
+
+## Standalone Mapper Plugins
+
+The other way of getting mapper functionality is to use a standalone mapper plugin that is inserted between your tap and target to do the translation `meltano run tap-csv <MAPPER> target-jsonl`.
+These can be used even whether or not your tap and/or target are SDK based, it's your choice how you implement mappers.
```
Comment (tayloramurphy) [inline docs/src/_guide/mappers.md:106]:
```suggestion
These can be used even when your your tap and/or target are not SDK based.
```

File: docs/src/_guide/mappers.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+The limitation with this approach is that currently you can only run these using the `meltano run` command.
```
Comment (tayloramurphy) [inline docs/src/_guide/mappers.md:108]:
```suggestion
The limitation with this approach is that currently you can only run these using the `meltano run` command and not using `meltano elt` or `meltano invoke`.
```

File: docs/src/_guide/mappers.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+## FAQs
+
+### Can I used standalone mapper plugins with the `meltano elt` command?
+
+No, currently only `meltano run` support standalone mapper plugins.
+
+### Can I use mapping features with the BATCH message type?
+
+No, BATCH message type does not currently support mapping functionality.
```
Comment (tayloramurphy) [inline docs/src/_guide/mappers.md:171]:
```suggestion
No, BATCH message type does not currently support mapping functionality.
Follow along with [this discussion](https://github.com/meltano/meltano/discussions/6639#discussioncomment-3417860) where we talk about how stream maps could support BATCH messages.
```

File: docs/src/_guide/mappers.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+An example taken from the [meltano-map-transformer](https://github.com/MeltanoLabs/meltano-map-transform/examples) repository shows the mapper altering casing of the data.
+Given this input CSV and mapper configurations:
+
+```csv
+id,first_name,last_name,email,ip_address
+1,Ethe,Book,ebook0@twitter.com,67.61.243.220
+```
+
+```yaml
+  - name: meltano-map-transformer
```
Comment (tayloramurphy) [inline docs/src/_guide/mappers.md:137]:
```suggestion
plugins
  mapper:
    - name: meltano-map-transformer
```

might be useful to give the context of where this is in the file.

**Review by tayloramurphy — APPROVED** [review]
@pnadolny13 minor comments, but this looks _great_. Coupled with the PR on the mapper repo, I'm quite happy with these updates.

File: docs/src/_guide/mappers.md (hunk 1 above)
Comment (pnadolny13) [reply docs/src/_guide/mappers.md]: @tayloramurphy I agree its confusing. I updated to column.

**Pull request merged**